core/src/test/scala/unit/kafka/coordinator/group/GroupMetadataTest.scala

```diff
@@ -120,32 +127,48 @@ class GroupMetadataTest extends JUnitSuite {
   }
 
   @Test(expected = classOf[IllegalStateException])
+  def testEmptyToStableIllegalTransition() {
+    group.transitionTo(Stable)
```

Comment:
I.e. you can also use `validateIllegalTransition` here and get rid of `expected=...`